int32 cmdId
string str_param
